hudi-snowflake/src/main/java/org/apache/hudi/snowflake/sync/SnowflakeSyncTool.java

```diff
@@ -0,0 +1,111 @@
+import org.apache.hudi.common.model.HoodieTableType;
+import org.apache.hudi.common.util.ValidationUtils;
+import org.apache.hudi.sync.common.AbstractSyncTool;
+import org.apache.hudi.sync.common.util.ManifestFileWriter;
+
+import com.beust.jcommander.JCommander;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+/**
+ * Tool to sync a hoodie table with a snowflake table. Either use it as an api
+ * SnowflakeSyncTool.syncHoodieTable(SnowflakeSyncConfig) or as a command line java -cp hoodie-hive.jar SnowflakeSyncTool [args]
+ * <p>
+ * This utility will get the schema from the latest commit and will sync snowflake table schema.
+ *
+ * @Experimental
+ */
+public class SnowflakeSyncTool extends AbstractSyncTool {
+
+  private static final Logger LOG = LogManager.getLogger(SnowflakeSyncTool.class);
+
+  public final SnowflakeSyncConfig cfg;
+  public final String stageName;
+  public final String manifestTableName;
+  public final String versionsTableName;
+  public final String snapshotViewName;
+
+  public SnowflakeSyncTool(TypedProperties properties, Configuration conf, FileSystem fs) {
+    super(properties, conf, fs);
+    cfg = SnowflakeSyncConfig.fromProps(properties);
+    stageName = cfg.tableName + "_stage";
+    manifestTableName = cfg.tableName + "_manifest";
+    versionsTableName = cfg.tableName + "_versions";
+    snapshotViewName = cfg.tableName;
+  }
+
+  @Override
+  public void syncHoodieTable() {
+    try (HoodieSnowflakeSyncClient snowSyncClient = new HoodieSnowflakeSyncClient(SnowflakeSyncConfig.fromProps(props), fs)) {
+      switch (snowSyncClient.getTableType()) {
+        case COPY_ON_WRITE:
+          syncCoWTable(snowSyncClient);
+          break;
+        case MERGE_ON_READ:
+        default:
+          throw new UnsupportedOperationException(snowSyncClient.getTableType() + " table type is not supported yet.");
+      }
+    } catch (Exception e) {
+      throw new HoodieSnowflakeSyncException("Got runtime exception when snowflake syncing " + cfg.tableName, e);
+    }
+  }
+
+  private void syncCoWTable(HoodieSnowflakeSyncClient snowSyncClient) {
+    ValidationUtils.checkState(snowSyncClient.getTableType() == HoodieTableType.COPY_ON_WRITE);
+    LOG.info("Sync hoodie table " + snapshotViewName + " at base path " + snowSyncClient.getBasePath());
+
+    ManifestFileWriter manifestFileWriter = ManifestFileWriter.builder()
+        .setConf(conf)
+        .setBasePath(cfg.basePath)
+        .setUseFileListingFromMetadata(cfg.useFileListingFromMetadata)
+        .setAssumeDatePartitioning(cfg.assumeDatePartitioning)
+        .build();
+    manifestFileWriter.writeManifestFile();
+
+    snowSyncClient.createStage(stageName, cfg.basePath, cfg.storageIntegration);
+    LOG.info("External temporary stage creation complete for " + stageName);
+    snowSyncClient.createManifestTable(stageName, manifestTableName);
```

Comment:
I am trying to understand how does snowSyncClient makes use of the manifest file that we just wrote. just the name would suffice is it? 
in BigQuerySync, I see we are fetching sourceUri (relative path) and passing it in to the client. here I don't see a good connection. can you help me understand.

Reply:
Please look at the createSnapshotView method, which is using the manifest file to filter the latest files from the versions table.